internal/extractors/innosat_reader.go

```diff
@@ -35,9 +41,11 @@ func DecodeSource(data []byte) (SourcePackage, error) {
 		return SourcePackage{}, err
 	}
 
+	sliceStart := binary.Size(header) + binary.Size(tmPayload)
+	sliceEnd := sliceStart + int(header.PacketLength) - binary.Size(tmPayload) - crcChecksumLength + pusLengthOffset
```

Comment:
This was not the error. The following line is better inline with the spec.
`sliceEnd := binary.Size(header) + int(header.PacketLength) - crcChecksumLength + pusLengthOffset`
even though
`sliceEnd := len(data) - crcChecksumLength `
still works.

Reply:
Not sure if you want me to change anything?